### PR TITLE
feat(outbound): add weighted random proxy-group selection

### PR DESCRIPTION
**Pull request by KagurazakaNyaa**
### Background

This change implements weighted random selection for proxy groups, without introducing a new policy name.

Previously, `policy: random` selected uniformly from alive nodes. This made it impossible to bias random selection toward preferred nodes while keeping random behavior. This patch adds `add_weight` as a filter annotation, similar to the existing `add_latency` annotation, so users can write configurations like:

```dae
filter: name(US_node) [add_weight: 4]
policy: random
```

The effective default weight is `1`, so `add_weight: 4` makes the node's effective weight `5`.

The change also covers related edge cases:

- zero-valued first annotations such as `add_latency: 0s` and `add_weight: 0`
- overflow around `1 + add_weight`
- aggregate overflow when selecting from groups with very large effective weights
- excluded or dead dialers should not be selected even when they have higher weights

In addition, it updates config marshal round-trip behavior so annotated `filter:` lines are preserved, and documents the actual first-match filter annotation semantics in the config description, example config, and docs.

### Checklist

- [x] The Pull Request has been fully tested
- [ ] There's an entry in the CHANGELOGS
- [x] There is a user-facing docs PR against https://github.com/daeuniverse/dae

### Full Changelogs

- Implement weighted random proxy-group selection via `add_weight` filter annotations
- Preserve annotated `filter:` lines during config marshal / round-trip
- Document first-match filter annotation semantics and weighted random usage in config descriptions, example config, and docs
- Fix zero-valued first-annotation handling and weight overflow edge cases
- Keep equal-weight random groups on the existing fast path
- Add regression tests for weighted random selection, dead/excluded dialers, and large aggregate weights

### Issue Reference

N/A

### Test Result

```sh
go test ./component/outbound/dialer
go test ./component/outbound
go test ./config
```

Output:

```text
ok      github.com/daeuniverse/dae/component/outbound/dialer    0.221s
ok      github.com/daeuniverse/dae/component/outbound           0.070s
ok      github.com/daeuniverse/dae/config                       0.075s
```

Full test suite note:

```sh
go test ./...
```

currently fails in existing eBPF-generated type references unrelated to this change, for example:

```text
control/control_plane_core.go:104:22: undefined: bpfObjects
control/udp_conn_state_tracker.go:12:14: undefined: bpfTuplesKey
```

### CODE REVIEW

File: example.dae [hunk 1]
```diff
@@ -323,6 +323,10 @@ group {
         #filter: name(node1, node2)
 
         # Filter nodes and give a fixed latency offset to archive latency-based failover.
+        # Filter lines are checked in order. If one node matches multiple filter lines,
+        # only the first matching line applies; later annotations are ignored.
+        # For example, if one node matches both `name(US_node) [add_latency: -500ms]`
+        # and `name(premium) [add_latency: 1000ms]`, only the first matching line applies.
         # In this example, there is bigger possibility to choose US node even if original latency of US node is higher.
```
[the remark this answers is not in the document]
Comment (KagurazakaNyaa) [reply example.dae]: @copilot apply changes based on [this feedback](https://github.com/daeuniverse/dae/pull/987#discussion_r3158408891)
